### DISCUSSION TITLE
修改docker镜像源配置后不会自动重启

### DISCUSSION THREAD

**Discussion by hjdyzy** [1Panel-dev/1Panel#5554] [Q&A]
### 联系方式

207581543@qq.com

### 1Panel 版本

v1.10.10-lts

### 问题描述

1、在 容器 > 配置 页面，修改镜像加速配置，删除镜像源后点击确定保存，会有个弹框让输入“立即重启”，输入完点确认后，docker实际上并没有重启；
2、只有新增数据源保存才会重启docker，但是重启完没有任何提示 说修改成功啥的；
功能请求：3、那个“立即重启”4个字能不能做成方面复制粘贴的那种，鼠标在文字上双击就直接选中这4个字了。

### 重现步骤

第1点：删除配置的docker镜像源，按照提示输入“立即重启”，点确定。
第2点：新增配置的docker镜像源，按照提示输入“立即重启”，点确定。

### 期待的正确结果

输入“立即重启”后，会重启docker；或者删除配置的docker镜像源点确定的时候就不要弹框让用户输入立即重启，就弹个提示说修改docker镜像源需要重启docker服务后生效

### 附加信息

![image](https://github.com/1Panel-dev/1Panel/assets/71299416/2db7cc4b-44d2-43d9-81fa-ba2a2179cab6)
![image](https://github.com/1Panel-dev/1Panel/assets/71299416/d23ca863-f081-46d3-a8ea-aa29b549ebdf)

**Comment by hjdyzy** [comment]
第2点经过多次测试后发现，要用反向代理的域名访问才会复现（关闭网站WAF情况下），测试只有把WAF全局关闭的情况下才会有成功的提示，只单独关1panel面板网站设置的WAF开关就会出现这个问题

**Comment by hjdyzy** [comment]
https://github.com/1Panel-dev/1Panel/assets/71299416/938b7ba8-2111-47b8-a2f8-7b848bf07e55

**Comment by wanghe-fit2cloud** [comment]
1、根据本地测试环境的验证，添加和修改镜像加速都需要重启操作。图中提供的信息显示，任务超时导致了重启失败。
2、操作成功后会有 `操作成功` 的提示信息
3、目前我们暂时没有计划支持该需求，望理解。
